### DISCUSSION TITLE
How to exclude some fields when getting List items

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#2001] [Q&A]
I have a collection `posts`. it has `content` field is rich editor so it's so heavy. so when I'm getting list too slow..

I have a question, how can I exclude the `content` field in the list?

**Comment by uteminhduc** [comment]
I have the same case as you. The temporary workaround is to use collection type `view`, [v0.13.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.13.0)
![image](https://user-images.githubusercontent.com/7042243/223618837-5b6ff378-a64e-4685-a240-285e33b5cb15.png)

**Reply by hungcrush** [reply]
Yeah I know that, but I want to improve in my Admin UI, it has to apply to all collections

**Comment by ganigeorgiev** [comment]
@hungcrush Hm, I'll investigate the list rendering issue in the Admin UI. Currently we show an excerpt of the text but still it needs to be parsed by the browser.

> I have a question, how can I exclude the content field in the list?

If you want to hide the field in the Admin UI listing, you can try with the "Toggle columns" table options (_the 3 dots in the top-right corner of the table_).

If you want to hide the field in the API response, then this at the moment is not possible (_it is in the [roadmap](https://github.com/orgs/pocketbase/projects/2?pane=issue&itemId=16153865)_).

**Reply by hungcrush** [reply]
> If you want to hide the field in the API response

@ganigeorgiev Have you ever thought there would be an option for each field `hide-in-list` ? 😬 Is it difficult to do?

**Reply by ganigeorgiev** [reply]
@hungcrush Server-side "hidden" fields for now are left out of the scope because it is not clear:

- who can see the field(s) - admins, users with manage rule access, some other custom criteria?
- who will be able to change the hidden field(s) (if allowed in the first place)?

As mentioned, client-side `?fields/select` support will be added sometime in the future, but if you want to hide fields server-side from the web API responses you can:

- Create a "View collection" with subset of the original collection fields
- Move the hidden/private fields to a separate collection with their own API rules and link back to the main collection via `relation`
- overwrite the API response(s) with the [app `*Request` event hooks](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks) (_this allow finer control but is also more involving_)

**Reply by hungcrush** [reply]
Thanks for your answer. I got it. 
I will use [*Request event hooks](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks) for my temporary plan
